contracts/ERC7432/ERC7432.sol

```diff
@@ -80,17 +81,26 @@ contract ERC7432 is IERC7432 {
         uint256 _tokenId,
         address _revoker,
         address _grantee
-    ) external override onlyApproved(_tokenAddress, _tokenId, _revoker) {
-        _revokeRole(_role, _tokenAddress, _tokenId, _revoker, _grantee);
+    ) external override {
+        bool _isRevokerApproved = _isRoleApproved(_tokenAddress, _tokenId, _revoker, msg.sender);
+        bool _isGranteeApproved = _isRoleApproved(_tokenAddress, _tokenId, _grantee, msg.sender);
+        require(_isRevokerApproved || _isGranteeApproved, "ERC7432: sender must be approved");
```

Comment:
I don't know what is the appropriate way of doing this in Solidity, but it's important for us to get it right to optimize for gas. There are four checks here, and you most likely do not need to execute all of them.

For example, if the first check returns true (which verifies if the sender is approved for all for the revoker), then you shouldn't need to verify the other 3 checks